### PR TITLE
Build tests

### PR DESCRIPTION
**Pull request by IngCr3at1on**
Only tested makefile.unix (on linux system)

Does not address failing tests, only makes them compile.
Several test files are disabled cause they broke the build,
will require further investigation.

### CODE REVIEW

**Comment by mitchellcash** [comment]
@IngCr3at1on you should add the Travis test builds to this pull as well.

**Comment by mitchellcash** [comment]
Some errors in key_tests.cpp

```
Opening LevelDB in /home/mitchell/.ion/txleveldb
Transaction index version is 70510
Opened LevelDB successfully
LoadBlockIndex(): hashBestChain=edde4f676c48e5c433a6cb9e93782506d1bfa6aa9e7a712332f8526db024de0d  height=41959  trust=133090380009911016336  date=07/02/16 10:40:32
LoadBlockIndex(): synchronized checkpoint 000001a7bb3214e3e1d2e4c256082b817a3c5dff5def37456ae16d7edaa508be
Verifying last 500 blocks at level 1
nFileVersion = 1000000
Keys: 0 plaintext, 0 encrypted, 0 w/ metadata, 0 total
Running 19 test cases...
test/key_tests.cpp(59): error in "key_test1": check bsecret1.SetString (strSecret1) failed
test/key_tests.cpp(60): error in "key_test1": check bsecret2.SetString (strSecret2) failed
test/key_tests.cpp(61): error in "key_test1": check bsecret1C.SetString(strSecret1C) failed
test/key_tests.cpp(62): error in "key_test1": check bsecret2C.SetString(strSecret2C) failed
test/key_tests.cpp(79): error in "key_test1": check addr1.Get() == CTxDestination(pubkey1.GetID()) failed
test/key_tests.cpp(80): error in "key_test1": check addr2.Get() == CTxDestination(pubkey2.GetID()) failed
test/key_tests.cpp(81): error in "key_test1": check addr1C.Get() == CTxDestination(pubkey1C.GetID()) failed
test/key_tests.cpp(82): error in "key_test1": check addr2C.Get() == CTxDestination(pubkey2C.GetID()) failed

*** 8 failures detected in test suite "Ion Test Suite"
```

**Comment by IngCr3at1on** [comment]
@mitchellcash as stated I did not address the failing tests, the goal of the included commit was ONLY to make them build and nothing more.

**Comment by mitchellcash** [comment]
Right, sorry! I was thinking this would have been taking out of the tests then, or is it required to build?

**Comment by mitchellcash** [comment]
In any event, confirmed linux builds. Will see if I can get my Mac builds working so I can test those

**Comment by IngCr3at1on** [comment]
@mitchellcash the ones that were renamed to `.bak` were to resolve the building but I didn't actually address the ones that were failing.

**Comment by mitchellcash** [comment]
@IngCr3at1on cool, cool. Thanks for that. Will report back after Mac build

**Comment by IngCr3at1on** [comment]
@mitchellcash presumably the key tests are failing because they're using bitcoin private keys instead of ion private keys ;)

File: src/makefile.osx [hunk 1]
```diff
@@ -156,10 +164,24 @@ obj/scrypt-x86_64.o: scrypt-x86_64.S
...
+	  rm -f $(@:%.o=%.d)
+
+test_ion: $(TESTOBJS) $(filter-out obj/init.o obj/bitcoin.o,$(OBJS:obj/%=obj/%))
+	$(CXX) $(CFLAGS) -o $@ $(LIBPATHS) $^ $(LIBS) $(TESTLIBS)
+
 clean:
-	-rm -f iond
+	-rm -f iond test_ion
 	-rm -f obj/*.o
+    -rm -f obj-test/*.o
```
Comment (mitchellcash) [inline src/makefile.osx]: Missing separator issue. You will need to fix the indentation.

File: src/makefile.osx [hunk 2]
```diff
@@ -156,10 +164,24 @@ obj/scrypt-x86_64.o: scrypt-x86_64.S
...
 	-rm -f obj/*.P
+    -rm -f obj-test/*.P
```
Comment (mitchellcash) [inline src/makefile.osx]: And here.

File: src/makefile.osx (hunk 1 above)
Comment (IngCr3at1on) [inline src/makefile.osx]: Atom being stupid about auto indention (well it's sort of Atom's fault); it's also the people who wrote the makefiles originally for using tabs and spaces in the same file.

Comment (mitchellcash) [inline src/makefile.osx]: Agreed! There's a strange mix.

**Pull request merged**